src/net.cpp

```diff
@@ -1434,6 +1435,20 @@ size_t V2Transport::GetSendMemoryUsage() const noexcept
     return sizeof(m_send_buffer) + memusage::DynamicUsage(m_send_buffer);
 }
 
+Span<const std::byte> V2Transport::GetSessionID() const noexcept
+{
+    if (m_use_v1) return m_v1_fallback.GetSessionID();
+    LOCK(m_recv_mutex);
+    if (m_recv_state == RecvState::V1) return m_v1_fallback.GetSessionID();
```

Comment:
9c3159378658c4b1535ab6c546512f7d1b2b3979: what's the difference between these two scenarios? `m_use_v1` vs `m_recv_state == RecvState::V1`?

Why not return `{}`? Unsafe / bad / not great?